File: src/main/java/com/example/project/service/PlacementService.java
package com.example.project.service;

import com.example.project.model.Placement;
import com.example.project.repository.PlacementRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class PlacementService {

    private final PlacementRepository placementRepository;

    public PlacementService(PlacementRepository placementRepository) {
        this.placementRepository = placementRepository;
    }

    public List<Placement> getAllPlacements() {
        return placementRepository.findAll();
    }

    public List<Placement> getPlacementsByYear(int year) {
        return placementRepository.findByYear(year);
    }
}

File: frontend/src/services/PlacementService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/placements/all';

const PlacementService = {
  getAllPlacements: () => axios.get(API_URL),
  getPlacementsByYear: (year) => axios.get(`${API_URL}/year/${year}`),
};

export default PlacementService;
